realm/realm-library/src/main/java/io/realm/OrderedRealmCollectionImpl.java

```diff
@@ -144,23 +143,15 @@ public E first(@Nullable E defaultValue) {
 
     @Nullable
     private E firstImpl(boolean shouldThrow, @Nullable E defaultValue) {
-        UncheckedRow row = collection.firstUncheckedRow();
-
-        if (forValues) {
-            // TODO implement this
-            return null;
-        }
-
-        if (row != null) {
-            //noinspection unchecked
-            return (E) realm.get((Class<? extends RealmModel>) classSpec, className, row);
-        } else {
+        if (collectionOperator.isEmpty()) {
             if (shouldThrow) {
                 throw new IndexOutOfBoundsException("No results were found.");
             } else {
                 return defaultValue;
             }
         }
+
+        return collectionOperator.get(0);
```

Comment:
I'm actually not a fan of this change after having to do optimizations specifically because `.size()` and `.get(0)` were slower than just accessing the first row directly. :|